packages/fe-fpm-writer/test/unit/filter.test.ts

```diff
@@ -0,0 +1,304 @@
+import type { Editor } from 'mem-fs-editor';
+import { create } from 'mem-fs-editor';
+import { create as createStorage } from 'mem-fs';
+import { join } from 'path';
+import type { CustomFilter } from '../../src/filter/types';
+import { generateCustomFilter } from '../../src/filter';
+import type { EventHandlerConfiguration, FileContentPosition } from '../../src/common/types';
+import { Placement } from '../../src/common/types';
+import os from 'os';
+
+describe('CustomFilter', () => {
+    describe('generateCustomFilter', () => {
+        const testDir = '' + Date.now();
+        let fs: Editor;
+        const filter: CustomFilter = {
+            name: 'NewCustomFilter',
+            label: 'Test Custom Filter',
+            controlID: 'testID',
+            property: 'Testing'
+        };
+        const getControllerPath = (controller: CustomFilter, isTs = false): string => {
+            return join(testDir, `webapp/${controller.folder}/${controller.name}.${isTs ? 'ts' : 'js'}`);
+        };
+        const getExpectedFragmentPath = (newFilter: CustomFilter): string =>
+            join(testDir, `webapp/${newFilter.folder}/${newFilter.fragmentFile}.fragment.xml`);
+
+        const getManifest = (extensions: unknown = undefined): string => {
+            const manifest = {
+                'sap.app': {
+                    id: 'myTestApp'
+                },
+                'sap.ui5': {
+                    dependencies: {
+                        libs: {
+                            'sap.fe.templates': {}
+                        }
+                    },
+                    routing: {
+                        targets: {
+                            TestListReport: { name: 'sap.fe.templates.ListReport' },
+                            TestObjectPage: { name: 'sap.fe.templates.ObjectPage' }
+                        }
+                    }
+                }
+            };
+            return JSON.stringify(manifest, null, 2);
+        };
+        const testAppManifest = getManifest();
+
+        beforeEach(() => {
+            fs = create(createStorage());
+            fs.delete(testDir);
+            fs.write(join(testDir, 'webapp/manifest.json'), testAppManifest);
+        });
+        test('New custom filter (no eventhandler)', () => {
+            generateCustomFilter(testDir, filter, fs);
+            expect(fs.readJSON(join(testDir, 'webapp/manifest.json'))).toMatchSnapshot();
+            expect(fs.exists(getControllerPath(filter))).toBeTruthy();
```

Comment:
See above comments. If `eventhandler` is undefined, then it shouldn't be created so this should be a `toBe(false)` check.

Reply:
Fixed with #1027